jsx/App/Stories/Story/Display/TextFormatButton.jsx

```diff
@@ -0,0 +1,243 @@
+
+/* Models a text format button. */
+export function TextFormatButton({ sentences, metadata }) {
+
+    /* 
+    Calls on individual helper functions to gather the morphemes,
+    gloss, translation, and title of the selected sentence. 
+    */
+    function processSentences() {
+        let dependents;
+        if (! sentences[0]) {
+            dependents = sentences["dependents"]
+        } else {
+            dependents = sentences[0]["dependents"];
+        }
+
+        const aingaeWordList = dependents[0]["values"];
+        const morphemeList = dependents[1]["values"];
+        const glossList = dependents[3]["values"];
+
+        const morphAndGloss = organizeWords(aingaeWordList, morphemeList, glossList);
+        const morphemeMap = morphAndGloss["morphemes"];
+        const glossMap = morphAndGloss["gloss"];
+        const sentenceTranslation = getSentenceTranslation(dependents);  
+
+        // Some metadata
+        const title = getTitle();
+        const storyId = getStoryId();
+        const sentenceUrl = getSentenceUrl();
+
+        return {
+            storyId : storyId,
+            title : title,
+            sentenceUrl : sentenceUrl,
+            morphemes : morphemeMap,
+            gloss : glossMap,
+            sentenceTranslation : sentenceTranslation  
+        };
+    }
+
+    /* Returns a map between each word and all of its sub-components (core and clitics, and gloss, etc.) */
+    function organizeWords(wordList, morphemeList, glossList) {
```

Comment:
A lot of this logic belongs in preprocessing, not in jsx. Preprocessing is when we have the most information about the original FLEx or ELAN file. Also, it's better for preprocessing to be slow than for the jsx to be slow. Preprocessing only happens once per update, whereas jsx happens every time a user loads the page. 

Adding to the preprocessing logic means we'll need to change or add to the json format that LingView uses to store files. All changes to LingView's json format need to be backwards-compatible, so I'd like to talk over the proposed changes before you put too much effort into coding and certainly before we merge them into BrownCLPS master.